### ISSUE TITLE
numpy.diff() broken for bool-Arrays

### ISSUE THREAD

**Issue by sscherfke** [numpy/numpy#9367]
In NumPy 1.13, `numpy.diff()` no longer works with Bool-Arrays:
```python
>>> import numpy as np
>>> np.diff([True, False, False])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python3.6/site-packages/numpy/lib/function_base.py", line 1926
, in diff
    return a[slice1]-a[slice2]
TypeError: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logi
cal_xor function instead.
```
It worked in NumPy 1.12.1 and the [documentation says](https://docs.scipy.org/doc/numpy/reference/generated/numpy.diff.html) that it should still work:
```python
>>> import numpy as np
>>> np.diff([True, False, False])
array([ True, False], dtype=bool)
```

**Comment by jaimefrio** [comment]
Thanks for the report!

This is actually a duplicate of #9251, which has already been fixed in #9259. So it should be fixed in current master, and in some upcoming bugfix release (1.13.1 I'm guessing).

Closing as already fixed, but please feel free to reopen or comment if you feel there's something missing.

**Comment by sscherfke** [comment]
Oh, this issue didn’t show up when I searched the issue for `diff`.  Nice to see that is has already been fixed. :)

**Comment by jaimefrio** [comment]
Yes, we are that good! ;-)

**Comment by bdowling** [comment]
Did this regress?
```
>>> df = pd.DataFrame({"C": [True, True, False, False, True, True]})

>>> df["C"].diff(1) != 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../lib/python3.7/site-packages/pandas/core/series.py", line 2193, in diff
    result = algorithms.diff(com.values_from_object(self), periods)
  File "/.../lib/python3.7/site-packages/pandas/core/algorithms.py", line 1819, in diff
    out_arr[res_indexer] = arr[res_indexer] - arr[lag_indexer]
TypeError: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.

>>> np.version.short_version
'1.15.4'
```

**Comment by bdowling** [comment]
Workaround:
```
>>> transitions[1:] = df["C"] ^ df["C"].shift().diff(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'transitions' is not defined
>>> transitions = df["C"] ^ df["C"].shift().diff(1)
>>> transitions[0] = False
>>> transitions
0    False
1     True
2    False
3     True
4     True
5    False
Name: C, dtype: bool
```

**Comment by bdowling** [comment]
Fix was in #9259

**Comment by bdowling** [comment]
Nevermind.  I realized after later that there is differences in df.diff() and np.diff()..

```
>>> df.loc[1:, 'diff'] = np.diff(df['C'])
>>> df
       C   diff
0   True    NaN
1   True  False
2  False   True
3  False  False
4   True   True
5   True  False
```

https://github.com/pandas-dev/pandas/issues/17294
